### PR TITLE
Windchild's Logger Expansion

### PR DESCRIPTION
**Pull request by Windchild292**
I've expanded upon @SJuliez's great logging changes in #2088 with this PR to handle the options missed in that, plus removed the pretty much unused StringBuilder versions. This will allow for significantly more simplification in logging functionality.

To Review: Whitespace differences off.

This MUST be merged at the SAME time as https://github.com/MegaMek/mekhq/pull/1963

### CODE REVIEW

**Review by SJuliez — APPROVED** [review]
Looks good. Just two suggestions.

And a question (but this isnt your code, I was just wondering) -  why do the late assignment:
```
MMLogger getLogger() {
        if (null == logger) {
            logger = DefaultMmLogger.getInstance();
        }
        return logger;
    }
```
... and not assign it directly as `MMLogger logger = DefaultMmLogger.getInstance();`?

Looking at it again, I wonder why I didnt notice that when taking the method name from the stack trace, the class name can be extracted from it as well. In this way methodbegin can be shortened to 
`logger.methodBegin();` When this is merged, I can add that. Actually this could shorten all the methods even more...

```
private String getCallingClass() {
        try {
            return Thread.currentThread().getStackTrace()[3].getClassName();
        } catch (Exception e) {
            return "DefaultMMLogger Error: Could not obtain class name";
        }
    }
```

File: megamek/src/megamek/MegaMek.java [hunk 1]
```diff
@@ -305,25 +303,22 @@ private static void startDedicatedServer(String[] args) {
...
-                    "args must be non-null");
+            getLogger().error(MegaMek.class, METHOD_NAME, "args must be non-null");
             return;
         }
         IMegaMekGUI mainGui = MegaMek.getGui(guiName);
         if (mainGui == null) {
-            getLogger().log(MegaMek.class, METHOD_NAME, LogLevel.INFO,
-                    UNKNOWN_GUI_MESSAGE + guiName);
+            getLogger().info(MegaMek.class, METHOD_NAME, UNKNOWN_GUI_MESSAGE + guiName);
             System.exit(1);
```
[the remark this answers is not in the document]
Comment (Windchild292) [reply megamek/src/megamek/MegaMek.java:316]: Yup, flipped.

File: megamek/src/megamek/client/bot/princess/MoralUtil.java [hunk 2]
```diff
@@ -131,7 +131,7 @@ public void checkMoral(boolean forcedWithdrawal, int bravery, int selfPreservati
                 }
             }
         } finally {
-            logger.log(getClass(), "checkMoral(boolean, int, int, IPlayer, IGame)", LogLevel.INFO, logMsg);
+            logger.log(getClass(), "checkMoral(boolean, int, int, IPlayer, IGame)", LogLevel.INFO, logMsg.toString());
```
[the remark this answers is not in the document]
Comment (Windchild292) [reply megamek/src/megamek/client/bot/princess/MoralUtil.java:134]: Yup, flipped.

**Comment by Windchild292** [comment]
I believe the late assignment is for unit testing purposes, based on notes within princess

**Comment by Windchild292** [comment]
Waiting on https://github.com/MegaMek/mekhq/pull/1963 to get this merged.

**Pull request merged**